phpBB/posting.php

```diff
@@ -29,9 +29,6 @@
 
 
 // Grab only parameters needed here
-$post_id	= $request->variable('p', 0);
-$topic_id	= $request->variable('t', 0);
-$forum_id	= $request->variable('f', 0);
```

Comment:
I don't understand why you can't keep the same logic here. With the removed parameters, the variables will still default to 0 as you're now doing starting at line 45 anyway.

Reply:
The intention was to ensure that the url parameters are never used except in the cases that they are needed, it was coded this way to show that the url parameters are used by exception (rather than reading them and overwriting them later).

Reply:
Totally reasonable IMO: Don't read the parameter before it is actually required. Assign with dummy value 0 only and read from request only when really required.